chain/chain/src/chain.rs

```diff
@@ -1013,6 +1013,18 @@ impl Chain {
         let shards_to_dl = self.get_shards_to_dl_state(me, &prev_hash);
         let prev_block = self.get_block(&prev_hash)?;
 
+        if prev_block.chunks().len() != block.chunks().len() && !shards_to_dl.is_empty() {
+            // Currently, the state sync algorithm assumes that the number of chunks do not change
+            // between the epoch being synced to and the last epoch.
+            // Fortunately, since all validators track all shards for now, this error will not be
+            // triggered in live yet
+            let str = format!(
+                "Cannot download states for epoch {:?} because sharding just changed. I'm {:?}",
+                block.header().epoch_id(),
+                me
+            );
+            return Err(Error::from(ErrorKind::Other(str)));
```

Comment:
I thought about it more. Instead of erroring here, I decided to simply log an error message. This way, the failure of this method will not affect the processing the the block for this epoch. So if this error is ever triggered, the validator will keep processing blocks until the end of this epoch and either halt or fail when the next epoch starts and it doesn't have the states ready for that.